File: app/src/main/java/com/conoland/pokedex/pokeapi/PokeapiService.java
package com.conoland.pokedex.pokeapi;


import com.conoland.pokedex.models.Pokemon;
import com.conoland.pokedex.models.PokemonRespuesta;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PokeapiService {

    @GET("pokemon")
    Call<PokemonRespuesta>obtenerListaPokemon(@Query("limit") int limit,@Query("offset") int offset);

    @GET("pokemon/{id}")
    Call<Pokemon>datos(@Path("id") int id);
}

File: app/src/main/java/com/conoland/pokedex/pokeapi/PokeapiService.kt
package com.conoland.pokedex
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import com.conoland.pokedex.models.Pokemon
import com.conoland.pokedex.models.PokemonRespuesta
import retrofit2.http.Query

interface PokeapiService {
    @GET("pokemon")
    fun obtenerListaPokemon(
        @Query("limit") limit: Int,
        @Query("offset") offset: Int
    ): Call<PokemonRespuesta>

    @GET("pokemon/{id}")
    fun datos(@Path("id") id: Int): Call<Pokemon>
}
